src/client/src/components/Modal/index.tsx

```diff
@@ -37,6 +37,7 @@ const getCustomStyles = (MODAL_TYPE: ModalType | undefined) => {
       left: 0,
       right: 0,
       bottom: 0,
+      zIndex: 2000,
```

Comment:
Just out of curiosity, what does this do?

Reply:
oh this just gives higher priority to the overlay div so that it overlaps all the other divs (including the header)

Reply:
Is there a specific reason for the number `2000` or it could be any number that is greater than other parts of the page?

Reply:
It's definitely the later :D Here is a nice strategy example: https://medium.com/science-journal/z-index-organization-in-css-5913fd4c25c9

Reply:
But the rule of thumb should be: we shouldn't go ridiculously high. 2000 seems to work here ;)